src/QMCWaveFunctions/Jastrow/J1Spin.h

```diff
@@ -166,8 +166,7 @@ struct J1Spin : public WaveFunctionComponent
         for (int j = 0; j < NumTargetGroups; j++)
         {
           auto igroup = Ions.getGroupID(i);
-          if (Ions.getGroupID(i) == source_type && j == target_type &&
-              J1UniqueFunctors[i * NumTargetGroups + j] == nullptr)
+          if (igroup == source_type && j == target_type && J1UniqueFunctors[igroup * NumTargetGroups + j] == nullptr)
```

Comment:
While you're making changes here, what do you think about moving line 168 and `if(igroup == source_type)`  out of the loop over `j`?